src/tablespace.c

```diff
@@ -315,47 +309,60 @@ tablespace_attach(PG_FUNCTION_ARGS)
 	tspcname = PG_GETARG_NAME(0);
 	hypertable_oid = PG_GETARG_OID(1);
 
+	tablespace_attach_direct(tspcname, hypertable_oid);
+
+	PG_RETURN_VOID();
+}
+
+void
+tablespace_attach_direct(Name tspcname, Oid hypertable_oid)
+{
+	Cache	   *hcache;
+	Hypertable *ht;
+	Oid			tspc_oid;
+	Oid			ownerid;
+	AclResult	aclresult;
+	CatalogSecurityContext sec_ctx;
+
 	tspc_oid = get_tablespace_oid(NameStr(*tspcname), true);
 
 	if (!OidIsValid(tspc_oid))
 		ereport(ERROR,
-				(errcode(ERRCODE_UNDEFINED_OBJECT),
-				 errmsg("Tablespace \"%s\" does not exist", NameStr(*tspcname)),
-				 errhint("A tablespace needs to be created"
-						 " before attaching it to a hypertable")));
+		(errcode(ERRCODE_UNDEFINED_OBJECT),
```

Comment:
What's with the indentation here? Pretty sure this was properly run through `pgindent` before.